ai_diffusion/ui/widget.py

```diff
@@ -633,6 +699,39 @@ def value(self, value: float):
             return
         self._slider.setValue(int(value * 100))
         self._input.setValue(int(value * 100))
+        self.update_suffix()
+
+    @property
+    def is_live(self):
+        return self._is_live
+
+    @is_live.setter
+    def is_live(self, is_live: bool):
+        self._is_live = is_live
+        self.update_spinbox()
+
+    def update_spinbox(self):
+        assert self.model
+        # copied from _sampling_from_style
```

Comment:
I think it's worth adding a function here that can be used in both places. Maybe `style.get_steps(is_live)` which returns `(min_steps, total_steps)`?

Reply:
done `style.get_steps`